src/core/server.cc

```diff
@@ -367,6 +376,49 @@ InferenceServer::UnloadModel(const std::string& model_name)
   return model_repository_manager_->LoadUnloadModel(model_name, action_type);
 }
 
+Status
+InferenceServer::RegisterSharedMemory(
+    const std::string& name, const std::string& shm_key, const size_t offset,
+    const size_t byte_size)
+{
+  if (ready_state_ != ServerReadyState::SERVER_READY) {
+    return Status(RequestStatusCode::UNAVAILABLE, "Server not ready");
+  }
+
+  ScopedAtomicIncrement inflight(inflight_request_counter_);
+
+  auto action_type = SharedMemoryManager::ActionType::REGISTER;
+  return shared_memory_manager_->RegisterUnregisterSharedMemory(
```

Comment:
Can shared memory manager just have separate register and unregister functions?

Reply:
Yes. There already are two separate functions. This is not necessary.